core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -1081,26 +1084,30 @@ private void traceRequest(AbstractRequest<?> request, TracedCommand command) {
       }
    }
 
-   void sendCommand(Address target, Object command, long requestId, DeliverOrder deliverOrder,
-                    boolean noRelay, boolean checkView) {
-      if (checkView && !clusterView.contains(target))
+   void sendCommandCheckingView(Address destination, Object command, long requestId, DeliverOrder deliverOrder) {
+      var target = clusterView.getAddressFromView(destination);
+      if (target == null) {
+         // not in view
          return;
+      }
+      sendCommand(target, command, requestId, deliverOrder, true);
+   }
 
-      Message message = new BytesMessage(toJGroupsAddress(target));
+   void sendCommand(org.jgroups.Address target, Object command, long requestId, DeliverOrder deliverOrder,
+                    boolean noRelay) {
+      Message message = new BytesMessage(target);
       marshallRequest(message, command, requestId);
       setMessageFlags(message, deliverOrder, noRelay);
 
       send(message);
       if (noRelay) {
+         assert target instanceof ExtendedUUID;
          // only record non cross-site messages
-         metricsManager.recordMessageSent(target, message.size(), requestId == Request.NO_REQUEST_ID);
+         // TODO FIX: the implementation uses `toString` from the address (uses the name in the tag)
```

Comment:
Should we create a GH issue to track this?

Reply:
https://github.com/infinispan/infinispan/issues/14744